tests/testthat/test-derive_params_growth_age.R

```diff
@@ -1,41 +1,398 @@
+  vs_data <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~VISIT, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "Study", "1001", "Screening", "M", 2, "months", "HEADC", 39,
+    "Study", "1002", "Cycle 1 Day 1", "F", 1157, "days", "HEADC", 50,
+  )
+
+  meta <- tibble::tribble(
+    ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
+    "M", 61, "days", 1, 39.1349, 0.02997,
+    "F", 1157, "days", 1, 48.6732, 0.02906,
+  )
+
+  actual <- derive_params_growth_age(
+    dataset = vs_data,
+    by_vars = exprs(STUDYID, USUBJID, VISIT),
+    sex = SEX,
+    age = AGECUR,
+    age_unit = AGEU,
+    meta_criteria = meta,
+    parameter = VSTESTCD == "HEADC",
+    analysis_var = VSSTRESN,
+    set_values_to_sds = exprs(
+      PARAMCD = "HDCASDS"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "HDCAPCTL"
+    )
+  )
+
+  vs_data_age <- vs_data %>%
+    mutate(
+      AGECUR_c = case_when(
+        AGEU == "days" ~ AGECUR,
+        AGEU == "months" ~ round(AGECUR * 30.4375)
+      ),
+      AGEU_c = "days"
+    ) %>%
+    mutate(AGE = AGECUR_c, AGEU = AGEU_c) %>%
+    select(-ends_with("_c"), -AGECUR)
+
+  vs_data_meta <- vs_data_age %>% inner_join(meta, by = c("SEX", "AGE", "AGEU"))
+
+  vs_sds <- vs_data_meta %>% mutate(AVAL = (((VSSTRESN / M)^L) - 1) / (L * S))
+  vs_pctl <- vs_sds %>% mutate(AVAL = pnorm(AVAL))
+  expected <- bind_rows(vs_sds, vs_pctl) %>% pull(AVAL)
+
+  vs_sds <- vs_data_meta %>% mutate(AVAL = (((VSSTRESN / M)^L) - 1) / (L * S))
+  vs_pctl <- vs_sds %>% mutate(AVAL = pnorm(AVAL))
+  expected <- bind_rows(vs_sds, vs_pctl) %>% pull(AVAL)
+
+
+  expect_equal(
+    filter(actual, PARAMCD %in% c("HDCASDS", "HDCAPCTL")) %>% pull(AVAL),
+    expected
+  )
+})
+
+## Test 5: Extreme BMI value derivation works ----
+test_that("derive_params_growth_age Test 5: Extreme BMI value derivation works", {
+  vs_data <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~VISIT, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "Study", "1001", "Screening", "M", 40.5, "months", "BMI", 19,
+  )
+
+  meta <- tibble::tribble(
+    ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S, ~P95, ~sigma,
+    "M", 1233, "days", -1.401671596, 15.85824093, 0.071691278, 18.0399, 2.022795,
+  )
+
+  actual <- derive_params_growth_age(
+    dataset = vs_data,
+    by_vars = exprs(STUDYID, USUBJID, VISIT),
+    sex = SEX,
+    age = AGECUR,
+    age_unit = AGEU,
+    meta_criteria = meta,
+    parameter = VSTESTCD == "BMI",
+    analysis_var = VSSTRESN,
+    set_values_to_sds = exprs(
+      PARAMCD = "BMIASDS"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "BMIAPCTL"
+    )
+  )
+
+  vs_data_age <- vs_data %>%
+    mutate(
+      AGECUR_c = case_when(
+        AGEU == "days" ~ AGECUR,
+        AGEU == "months" ~ round(AGECUR * 30.4375)
+      ),
+      AGEU_c = "days"
+    ) %>%
+    mutate(AGE = AGECUR_c, AGEU = AGEU_c) %>%
+    select(-ends_with("_c"), -AGECUR)
+
+  vs_data_meta <- vs_data_age %>% inner_join(meta, by = c("SEX", "AGE", "AGEU"))
```

Comment:
I suspect the `inner_join()` call here is why you needed to put `library(dplyr)` at the top, you can either do `dplyr::inner_join()` or just put `inner_join()` into the `importFrom`

Reply:
Thanks @zdz2101 ! Will update in the next commit.